component/kafka/group/option.go

```diff
@@ -26,6 +29,52 @@ func FailureStrategy(fs kafka.FailStrategy) OptionFunc {
 	}
 }
 
+// CheckTopic will attempt to:
+//	1. connect to the broker,
+//  2. retrieve the existing topics in the broker
+//  3. if auto-topic creation is disabled, check whether the configured topics exist in the broker
+// If any of the checks above fail the component will exit before starting to consume messages
+func CheckTopic() OptionFunc {
+	return func(c *Component) error {
+		c.topicChecker = func(ctx context.Context, c *Component) error {
+			client, err := sarama.NewClient(c.brokers, c.saramaConfig)
+			defer func() {
+				if client != nil {
+					_ = client.Close()
+				}
+			}()
+			// without client no connection can be established
+			if err != nil {
+				return err
+			}
+			// check if the topic exists
+			brokerTopics, err := client.Topics()
+			if err != nil {
+				return err
+			}
+
+			// if auto-topic creation is not enabled then check if the topic exists
+			if !c.saramaConfig.Metadata.AllowAutoTopicCreation {
+				for _, topic := range c.topics {
```

Comment:
I know concretely it won't change a lot but having a O(n²) time complexity huts me a little bit. Couldn't we for example convert the current list of topics inside a map and make the checks from this map?

Reply:
Was the sense of my reaction in Slack but I feel like it wasn't clear enough :p
![image](https://user-images.githubusercontent.com/934784/146598337-77b75cff-f0ca-4e85-9a35-2b2096357b28.png)

Reply:
I completely misread this one. ;)

Reply:
hahahah never would have read O(n^2) in that comment... 

:D 

I can transform that as well. One small note though: this is actually O(n*m) where n is the amount of topics in the brokers and m is the amount of topics configured locally -> I expect that in most cases m << n and in any case as you know, cycling m times on the n-topics is irrelevant compared to the network roundtrip to the broker.